inbm/dispatcher-agent/tests/unit/source/test_ubuntu_source_cmd.py

```diff
@@ -279,6 +279,20 @@ def test_successfully_remove_gpg_key_and_source_list(
         except SourceError:
             self.fail("Remove GPG key raised DispatcherException unexpectedly!")
 
+
+     def test_successfully_add_gpg_key(self):
```

Comment:
let's have a positive and a negative test -- one where adding the key succeeds due to it being in the trusted repos list, and one where adding the key fails.